src/app/Routes.tsx

```diff
@@ -64,101 +65,103 @@ const Routes = (): JSX.Element => (
       <Route
         key={path}
         path={path}
-        render={() => (
+        element={
           <ErrorBoundary>
             <Controllers />
           </ErrorBoundary>
-        )}
+        }
       />
     ))}
-    {[devicesURLs.devices.index, devicesURLs.device.index(null, true)].map(
-      (path) => (
-        <Route
-          key={path}
-          path={path}
-          render={() => (
-            <ErrorBoundary>
-              <Devices />
```

Comment:
Instead of wrapping the list and details components in the same component we'd need to move all top level paths (e.g. `/devices/`, `/device/`) into this file. These top level components (e.g. `Devices`) don't currently contain any state, they just wrap the routes.